doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/docUtils.kt

```diff
@@ -0,0 +1,446 @@
+fun DocContent.findTagNamesInDocContent(): List<String> =
+    findInlineTagNamesInDocContent() + findBlockTagNamesInDocContent()
+
+
+/** Is able to find an entire JavaDoc/KDoc comment including the starting indent. */
+val docRegex = Regex("""( *)/\*\*([^*]|\*(?!/))*?\*/""")
+
+// Regex to match [Aliased][ReferenceLinks].
+val aliasedLinkRegex = Regex("""(\[[^\[\]]*]\[)([^\[\]]*)(])""")
+
+// Regex to match [ReferenceLinks].
+val singleLinkRegex = Regex("""((^|[^]])\[)([^\[\]]*)(]$|][^\[])""")
+
+// Regex to match [ReferenceLinks] [with space] since this breaks singleLinkRegex.
+val doubleSingleLinkRegex = Regex("""((^|[^]])\[)([^\[\]]*)(][^\[]\[)([^\[\]]*)(]$|][^\[])""")
+
+val javaLinkRegex = Regex("""\{@link.*}""")
+
+/**
+ * Replace KDoc links in doc content with the result of [process].
+ *
+ * Replaces all `[Aliased][ReferenceLinks]` with `[Aliased][ProcessedPath]`
+ * and all `[ReferenceLinks]` with `[ReferenceLinks][ProcessedPath]`.
+ */
+fun DocContent.replaceKdocLinks(process: (String) -> String): DocContent = this
+    .replace(aliasedLinkRegex) { // replace all [Aliased][ReferenceLinks] with [Aliased][ProcessedPath]
+        it.groupValues.let {
+            buildString {
+                append(it[1])
+                append(
+                    process(it[2])
+                )
+                append(it[3])
+            }
+        }
+    }
+    .replace(doubleSingleLinkRegex) { // replace all [ReferenceLinks] [with space] with [ReferenceLinks][ProcessedPath] [with space][ProcessedPath]
+        it.groupValues.let {
+            buildString {
+                append(it[1])
+                append(it[3])
+                append("][")
+                append(
+                    process(it[3])
+                )
+                append(it[4])
+                append(it[5])
+                append("][")
+                append(
+                    process(it[5])
+                )
+                append(it[6])
+            }
+        }
+    }
+    .replace(singleLinkRegex) { // replace all [ReferenceLinks] with [ReferenceLinks][ProcessedPath]
+        it.groupValues.let {
+            buildString {
+                append(it[0].dropLastWhile { it != ']' })
+                append("[")
+                append(
+                    process(it[3])
+                )
+                append(it[4])
+            }
+        }
+    }
+
+
+/**
+ * Finds removes the last occurrence of [element] from the list and, if found, all elements after it.
+ * Returns true if [element] was found and removed, false otherwise.
+ */
+fun <T> MutableList<T>.removeAllElementsFromLast(element: T): Boolean {
+    val index = lastIndexOf(element)
+    if (index == -1) return false
+    val indicesToRemove = index..lastIndex
+    for (i in indicesToRemove.reversed()) {
+        removeAt(i)
+    }
+    return true
+}
+
+// used to keep track of the current blocks
+private const val CURLY_BRACES = '{'
```

Comment:
I don't know why, but I prefer constants at the top of the file or in the separate file. Probably it's not sufficient